x-pack/solutions/observability/plugins/observability_logs_explorer/public/redirect_to_discover.tsx

```diff
@@ -27,7 +28,35 @@ export const renderDiscoverRedirect = (core: CoreStart, appParams: AppMountParam
 
 export const DiscoverRedirect = ({ core }: { core: CoreStart }) => {
   const location = useLocation();
-  const path = `${location.pathname}${location.search}`;
+  const searchParams = new URLSearchParams(location.search);
+  const pageStateEncoded = searchParams.get('pageState') || '';
+  const pageState = (safeDecode(pageStateEncoded) || {}) as {
```

Comment:
This is some impressive reverse-engineering of the url state. If we wanted to preserve type safety and ensure full compatibility we could resurrect the actual schema used for the url state from the `8.x` branch before the removal of the app: https://github.com/elastic/kibana/blob/4396d3a2ce5b1ff832405799c7d7676c564c7341/x-pack/solutions/observability/plugins/observability_logs_explorer/public/state_machines/observability_logs_explorer/src/url_state_storage_service.ts#L51-L61

What do you think?